app/src/androidTest/java/com/h/pixeldroid/MockedServerTest.kt

```diff
@@ -501,5 +516,71 @@ class MockedServerTest {
         onView(first(withId(R.id.commentContainer)))
             .check(matches(hasDescendant(withId(R.id.comment))))
     }
+
+    @Test
+    fun clickOnSensitiveWarningHidesIt() {
+
+        activityScenario.onActivity { it ->
+            assert(it.sensitiveWarning.visibility == VISIBLE)
+            it.sensitiveWarning.performClick()
+            Thread.sleep(1000)
+            assert(it.sensitiveWarning.visibility == VISIBLE)
+
+        }
+    }
+
+    @Test
+    fun clickOnImageViewHidesSensitiveWarning() {
+
+        activityScenario.onActivity { it ->
+            assert(it.sensitiveWarning.visibility == VISIBLE)
+            it.postPicture.performClick()
+            Thread.sleep(1000)
+            assert(it.sensitiveWarning.visibility == GONE)
+
+        }
+    }
+
+    @Test
+    fun censorMatrices() {
+        val array: FloatArray = floatArrayOf(
+            0.1f, 0f, 0f, 0f, 0f,  // red vector
+            0f, 0.1f, 0f, 0f, 0f,  // green vector
+            0f, 0f, 0.1f, 0f, 0f,  // blue vector
+            0f, 0f, 0f, 1f, 0f   ) // alpha vector
+
+        assert(censorColorMatrix().equals(array))
+        assert(uncensorColorMatrix().equals(ColorMatrix()))
+    }
+
+    @Test
+    fun performClickOnSensitiveWarning() {
+
+        onView(withId(R.id.list)).perform(slowSwipeUp(false))
+        onView(withId(R.id.list)).perform(slowSwipeUp(false))
+        Thread.sleep(1000)
+
+        onView(withId(R.id.list))
+            .perform(actionOnItemAtPosition<PostViewHolder>
+                (1, clickChildViewWithId(R.id.sensitiveWarning)))
+        Thread.sleep(1000)
+
+        assert(true)
+    }
+
+    @Test
+    fun performClickOnPostPicture() {
+
+        onView(withId(R.id.list)).perform(slowSwipeUp(false))
+        onView(withId(R.id.list)).perform(slowSwipeUp(false))
+        Thread.sleep(1000)
+
+        onView(withId(R.id.list))
+            .perform(actionOnItemAtPosition<PostViewHolder>
+                (1, clickChildViewWithId(R.id.postPicture)))
+        Thread.sleep(1000)
+
+        assert(true)
```

Comment:
Try testing something in your tests please, like if the post has actually been shown or not.

Reply:
Commit : e93fdff